service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/util/SectionFilterer.java

```diff
@@ -0,0 +1,52 @@
+package uk.nhs.adaptors.oneoneone.cda.report.util;
+
+import org.apache.commons.lang3.ArrayUtils;
+import uk.nhs.connect.iucds.cda.ucr.CE;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component5;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Section;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SectionFilterer {
+
+    private List<String> filterCodes = new ArrayList<>();
+
+    public SectionFilterer(List<String> filterCodes) {
+        this.filterCodes = filterCodes;
+    }
+
+    public List<POCDMT000002UK01Section> findValidSections(POCDMT000002UK01Section section) {
+
+        // Base case: If there are no nested sub components
+        if (ArrayUtils.isEmpty(section.getComponentArray())) {
```

Comment:
section.sizeOfComponentArray() ?

Reply:
(Let's discuss) Anyway, shouldn't stream simply be empty for empty collection or array? So if size o fsection.getComponentArray() is 0, then list of subSections should be empty. Id subSections list is empty, then valid Sections list also will be empty. So do we need explicitly return empty collection for empty component array?